ui/pages/token-allowance/token-allowance.js

```diff
@@ -94,9 +97,21 @@ export default function TokenAllowance({
   const unapprovedTxCount = useSelector(getUnapprovedTxCount);
   const unapprovedTxs = useSelector(getUnapprovedTransactions);
 
-  const customPermissionAmount = customTokenAmount.toString();
+  const replaceCommaToDot = (inputValue) => {
+    return inputValue.replace(/,/gu, '.');
+  };
+
+  let customPermissionAmount = replaceCommaToDot(customTokenAmount).toString();
+
+  const maxTokenAmount = calcTokenAmount(MAX_TOKEN_ALLOWANCE_AMOUNT, decimals);
+  if (customTokenAmount.length > 1 && Number(customTokenAmount)) {
+    const customSpendLimitNumber = new BigNumber(customTokenAmount);
+    if (customSpendLimitNumber.greaterThan(maxTokenAmount)) {
+      customPermissionAmount = 0;
+    }
+  }
 
-  const customTxParamsData = customTokenAmount
+  const customTxParamsData = Number(replaceCommaToDot(customTokenAmount))
```

Comment:
can we just check `customPermissionAmount` here? instead of `Number(replaceCommaToDot(customTokenAmount))`?

Reply:
Yes we can @danjm! Thank you! Done [here](https://github.com/MetaMask/metamask-extension/pull/16676/commits/b86cffca83aec2a922db38f8e7bb36936cf9ac0d).